### PR TITLE
Tweak compilation cache for SQL queries

### PR DESCRIPTION
**Pull request by aljazerzen**
- we had `QueryUnit.cachable` set to False for sql-over-edgedb-protocol,
- we were using the hash of the whole `NormalizedSource` as cache key, which includes extracted constants, which defeats (most) of the purpose of caching

Commits:
- **fix NormalizedSource.variables**
- **compute cache_key from pg_query_fingerprint**
- **Revert "compute cache_key from pg_query_fingerprint"**
- **pgsql Source.cache_key**
- **more tests**

### CODE REVIEW

**Comment by msullivan** [comment]
Any idea what's going on with the test failure?

**Comment by aljazerzen** [comment]
We were rejecting multi-statement queries in execute, after trying to fetch type descriptors from PostgreSQL.

File: edb/pgsql/parser/parser.pyx [hunk 1]
```diff
@@ -145,6 +145,7 @@ def pg_normalize(query: str) -> NormalizedQuery:
             raise error
 
         normalized_query = result.normalized_query.decode('utf8')
+        print("normalized_query", normalized_query)
```
Comment (msullivan) [inline edb/pgsql/parser/parser.pyx:148]: Drop this

File: edb/testbase/server.py [hunk 2]
```diff
@@ -401,6 +402,19 @@ def __getstate__(self):
...
+        self, measure_fn: typing.Callable[[], int | float],
+        expected_change: int | float
+    ):
+        before = measure_fn()
+        try:
+            yield
+        finally:
+            after = measure_fn()
+            change = after - before
+            self.assertEqual(expected_change, change)
```
Comment (msullivan) [inline edb/testbase/server.py:416]: I like this

**Review by msullivan — APPROVED** [review]

**Pull request merged**